src/DeviceManager.Api/Data/Management/ContextFactory.cs

```diff
@@ -71,22 +73,51 @@ private DbContextOptionsBuilder<DeviceContext> ChangeDatabaseNameInConnectionStr
             ValidateDefaultConnection();
 
             // 1. Create Connection String Builder using Default connection string
-            var sqlConnectionBuilder = new SqlConnectionStringBuilder(this.connectionOptions.Value.DefaultConnection);
+            // var sqlConnectionBuilder = new SqlConnectionStringBuilder(this.connectionOptions.Value.DefaultConnection);
+            var connectionBuilder = GetConnectionBuilder();
 
             // 2. Remove old Database Name from connection string
-            sqlConnectionBuilder.Remove(DatabaseFieldKeyword);
+            connectionBuilder.Remove(DatabaseFieldKeyword);
 
             // 3. Obtain Database name from DataBaseManager and Add new DB name to 
-            sqlConnectionBuilder.Add(DatabaseFieldKeyword, this.dataBaseManager.GetDataBaseName(tenantId));
+            connectionBuilder.Add(DatabaseFieldKeyword, this.dataBaseManager.GetDataBaseName(tenantId));
 
             // 4. Create DbContextOptionsBuilder with new Database name
             var contextOptionsBuilder = new DbContextOptionsBuilder<DeviceContext>();
 
-            contextOptionsBuilder.UseSqlServer(sqlConnectionBuilder.ConnectionString);
+            //contextOptionsBuilder.UseSqlServer(sqlConnectionBuilder.ConnectionString);
+            SetConnectionString(contextOptionsBuilder, connectionBuilder.ConnectionString);
 
             return contextOptionsBuilder;
         }
 
+        /// <summary>
+        /// Get Connection string builder instance based on the database type
+        /// </summary>
+        /// <returns></returns>
+        private DbConnectionStringBuilder GetConnectionBuilder()
+        {
+            if (connectionOptions.Value.UseNoSql)
```

Comment:
In my opinion, here, would be better to use ternary operator `?:` like  `(condition ? first_expression : second_expression; )` What do you think ?

Reply:
Yeah, I will fix all the changes you pointed out :-) 
By the way when are you planning to release next version of the framework??

Reply:
Hey, I would like to do it after I merge this PR ;)